docs/how-to-guides/effective-gno.md

```diff
@@ -0,0 +1,652 @@
+retrieve what you need, similar to SQL indexes.
+
+By using `avl.Tree` and other efficient data structures, you can optimize your
+Gno code for performance and cost-effectiveness, making your applications more
+scalable and efficient.
+
+TODO: multi-indices example
+
+### Construct "safe" objects
+
+A safe object in Gno is an object that is designed to be tamper-proof and
+secure. It's created with the intent of preventing unauthorized access and
+modifications. This follows the same principle of making a package an API, but
+for a Go object that can be directly referenced by other realms.
+
+The goal is to create an object which, once instantiated, can be linked and its
+pointer can be "stored" by other realms without issue, because it protects its
+usage completely.
+
+```go
+type MySafeStruct {
+    counter nb
+    admin std.Address
+}
+
+func NewSafeStruct() *MySafeStruct {
+    caller := std.GetOrigCaller()
+    return &MySafeStruct{
+        counter: 0,
+        admin: caller,
+    }
+}
+
+func (s *MySafeStruct) Counter() int { return s.counter }
+func (s *MySafeStruct) Inc() {
+    caller := std.GetOrigCaller()
+    if caller != s.admin {
+        panic("permission denied")
+    }
+    s.counter++
+}
+```
+
+Then, you can register this object in another or several other realms so other
+realms can access the object, but still following your own rules.
+
+```go 
+import "gno.land/r/otherrealm"
+
+func init() {
+    mySafeObj := NewSafeStruct()
+    otherrealm.Register(mySafeObject)
+}
+
+// then, otherrealm can call the public functions but won't be the "owner" of
+// the object.
+```
+
+### Choosing between native tokens and GRC20 tokens
+
+In Gno, you've got two main choices for tokens: Native or GRC20. Each has its
+own pros and cons, and the best fit depends on your needs.
+
+#### Native tokens
+
+Native tokens are managed by the banker module, separate from Gnovm. They're
+simple, strict, and secure. You can create, transfer, and check balances with an
+RPC call, no Gnovm needed.
+
+For example, if you're creating a coin for cross-chain transfers, native tokens
+are your best bet. They're IBC-ready and their strict rules offer top-notch
+security.
+
+TODO: native banker example.
+
+#### GRC20 tokens
+
+GRC20 tokens, on the other hand, are like Ethereum's ERC20 or CosmWasm's CW20.
+They're flexible, composable, and perfect for DeFi protocols and DAOs. They
+offer more features like token-gating, vaults, and wrapping.
+
+For instance, if you're creating a voting system for a DAO, GRC20 tokens are
+ideal. They're programmable, can be embedded in safe Go objects, and offer more
+control.
+
+Remember, GRC20 tokens are more gas-intensive and aren't IBC-ready yet. They
+also come with shared ownership, meaning the contract retains some control.
+
+In the end, your choice depends on your needs: simplicity and security with
+Native tokens, or flexibility and control with GRC20 tokens. And if you want the
```

Comment:
Hum, I’ve a doubt, because we’re using  `std.Coin` internally…

Probably coin should be the consistent wording, and token should be referenced in a “terms” section or doc.